### ISSUE TITLE
[ts-lit-plugin] Types from `vscode-html-languageservice`

### ISSUE THREAD

**Issue by ptu14** [runem/lit-analyzer#297]
Hi, I'm wondering why `lit-analyzer` that `ts-lit-plugin` using, uses type definitions from `vscode-html-languageservice` rather than `typescript` lib.dom or own type definitions.

For example `role` attribute case:

When i create Lit component with native input inside:

```typescript
	 render() {
		return html`
			<input role="${ifDefined(this.role || undefined)}" />
		`;
	}
```

lit-plugin in vscode and in Intellij highlight this as type error

![image](https://user-images.githubusercontent.com/20875935/220115229-2fc8825f-6186-4d6a-ab14-be6595069615.png)

when i run `npx lit-analyzer`

i get 
```bash
    Type 'string' is not assignable to '"alert" | "alertdialog" | "button" | "checkbox" | "dialog" | "gridcell" | "link" | "log" | "marquee" | "menuitem" | "menuitemcheckbox" | "menuitemradio" | "option" | "progressbar" | "radio" | "scrollbar" | "searchbox" | "slider" | "spinbutton" | "status" | "switch" | ... 89 more ...'
    138:  <input role="${ifDefined(this.role
    no-incompatible-type-binding

```

but inside `typescript/lib/lib.dom.d.ts` we have `role: string | null;` 

I understand that Lit doesn't always have a connection to TypeScript, but what it's doing now is also not accurate according to https://w3c.github.io/aria/#ARIAMixin. where type of `role` is `DOMString`.

maybe this bug should be reported to `ts-lit-plugin`, but from what I can see these projects are connected somehow.

**Comment by wszydlak** [comment]
Hi, this is not only connected to role property. This seems to be a problem with all ariamixin related attributes, eg:

<img width="687" alt="image" src="https://user-images.githubusercontent.com/6583977/220127187-c6f736ff-7476-4ab4-a998-8db8df392248.png">

**Comment by wszydlak** [comment]
I noticed that there is already reported issue with `autocomplete`: #267

But in this case also it should be `string`: https://html.spec.whatwg.org/multipage/input.html#htmlinputelement

**Comment by jesseditson** [comment]
FWIW this also appears to be true for all input, select, etc types from `lib.dom.d.ts` - for instance, a similar error will be reported when using:

```typescript
const value: "hidden" | "text" = "hidden";
html`<input
    type={value as HTMLInputElement["type"]}
/>
```

This makes it quite hard to get both lit analyzer and typescript types passing on projects that use both.
